### PR TITLE
apimachinery/meta/types.go: define InitialEventsListBlueprintAnnotationKey const

### PR DESCRIPTION
**Pull request by p0lyn0mial**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
```
// InitialEventsListBlueprintAnnotationKey is the name of the key
// where an empty, base64-encoded, versioned list is stored as a string.
//
// Used by clients that want to reconstruct a list object from
// the retrieved stream, ensuring it is equivalent to a standard LIST call.
//
// The annotation is added to a "Bookmark" event.
```
See also: https://github.com/kubernetes/enhancements/pull/4822

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
xref: https://github.com/kubernetes/enhancements/issues/3157

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go [hunk 1]
```diff
@@ -439,6 +439,15 @@ const (
...
 	InitialEventsAnnotationKey = "k8s.io/initial-events-end"
+
+	// InitialEventsListBlueprintAnnotationKey is the name of the key
+	// where an empty, base64-encoded, versioned list is stored as a string.
+	//
+	// Used by clients that want to reconstruct a list object from
+	// the retrieved stream, ensuring it is equivalent to a standard LIST call.
+	//
+	// The annotation is added to a "Bookmark" event.
+	InitialEventsListBlueprintAnnotationKey = "kubernetes.io/initial-events-list-blueprint"
```
Comment (wojtek-t) [inline staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go:450]: I'm conceptually fine with the approach, but how exactly the annotation should be named and whether the comment is convincing enough, I would leave to @deads2k

File: staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go [hunk 2]
```diff
@@ -439,6 +439,15 @@ const (
 	//
 	// The annotation is added to a "Bookmark" event.
 	InitialEventsAnnotationKey = "k8s.io/initial-events-end"
+
+	// InitialEventsListBlueprintAnnotationKey is the name of the key
+	// where an empty, base64-encoded, versioned list is stored as a string.
```
Comment (deads2k) [inline staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go:444]: explain that the encoding matches the request encoding and can therefore be proto, json, cbor, (can it be the table format?)

File: staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go (hunk 1 above)
Comment (deads2k) [reply staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go:450]: ref for those wondering wy kubernetes.io: https://github.com/kubernetes/community/blob/e00e0ada0c01d950debd196af7e96cc3218adf70/contributors/devel/sig-architecture/api-conventions.md?plain=1#L1858-L1859

> - "kubernetes.io" is the preferred form for labels and annotations, "k8s.io" should not be used
      for new map keys.

past mistakes do not justify future mistakes.

File: staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go (hunk 2 above)
Comment (p0lyn0mial) [reply staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go]: done, ptal.

**Pull request merged**